llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -5373,6 +5371,18 @@ bool InstCombinerImpl::prepareWorklist(Function &F) {
   return MadeIRChange;
 }
 
+void InstCombiner::computeBackEdges() {
+  // Collect backedges.
+  SmallPtrSet<BasicBlock *, 16> Visited;
+  for (BasicBlock *BB : RPOT) {
+    Visited.insert(BB);
+    for (BasicBlock *Succ : successors(BB))
+      if (Visited.contains(Succ))
+        BackEdges.insert({BB, Succ});
```

Comment:
```suggestion
      if (Succ != BB && Visited.contains(Succ))
        BackEdges.insert({BB, Succ});
```
Can we avoid inserting trivial edges `BB->BB`?

Reply:
We can, but I don't see a difference in compile-time from adding this special case.

Reply:
I'd say not having an extra special case is preferable then.